### PR TITLE
CLI to interate and calculate file checksums from destination directory

### PR DESCRIPTION
**Pull request by prashant-sb**
```
prashant@thinkpad:file_signatures$ ./runner -dest=/home/prashant/workspace/go-utils -sign=md5
/home/prashant/workspace/go-utils/.git/COMMIT_EDITMSG :: c290ec9ded96a3c67ae1ae669949645b
/home/prashant/workspace/go-utils/.git/HEAD :: 5536e138906060f13048b0f5f7d7313f
/home/prashant/workspace/go-utils/.git/config :: 6568beffa63e0bf3a5eb1922bf2b6bdd
/home/prashant/workspace/go-utils/.git/description :: a0a7c3fff21f2aea3cfa1d0316dd816c
/home/prashant/workspace/go-utils/.git/hooks/applypatch-msg.sample :: ce562e08d8098926a3862fc6e7905199
/home/prashant/workspace/go-utils/.git/hooks/commit-msg.sample :: 579a3c1e12a1e74a98169175fb913012
/home/prashant/workspace/go-utils/.git/hooks/fsmonitor-watchman.sample :: db4d62db2039e4bec38baa7567788284
/home/prashant/workspace/go-utils/.git/hooks/post-update.sample :: 2b7ea5cee3c49ff53d41e00785eb974c
/home/prashant/workspace/go-utils/.git/hooks/pre-commit.sample :: e4db8c12ee125a8a085907b757359ef0
/home/prashant/workspace/go-utils/.git/hooks/pre-applypatch.sample :: 054f9ffb8bfe04a599751cc757226dda
/home/prashant/workspace/go-utils/.git/hooks/pre-merge-commit.sample :: 39cb268e2a85d436b9eb6f47614c3cbc
/home/prashant/workspace/go-utils/.git/hooks/pre-push.sample :: 3c5989301dd4b949dfa1f43738a22819
/home/prashant/workspace/go-utils/.git/hooks/pre-rebase.sample :: 56e45f2bcbc8226d2b4200f7c46371bf
/home/prashant/workspace/go-utils/.git/hooks/pre-receive.sample :: 2ad18ec82c20af7b5926ed9cea6aeedd
/home/prashant/workspace/go-utils/.git/hooks/prepare-commit-msg.sample :: 2b5c047bdb474555e1787db32b2d2fc5
/home/prashant/workspace/go-utils/.git/hooks/update.sample :: 517f14b9239689dff8bda3022ebd9004
/home/prashant/workspace/go-utils/.git/index :: c9a071f46fedd75454a8eb5db5bf7f85
/home/prashant/workspace/go-utils/.git/info/exclude :: 036208b4a1ab4a235d75c181e685e5a3
/home/prashant/workspace/go-utils/.git/logs/HEAD :: f3ce3e76749ebdab62de3f61c2261fcf
/home/prashant/workspace/go-utils/.git/logs/refs/heads/go_filehash :: 319b66cab571ca33ae9803260a6dfbdf
/home/prashant/workspace/go-utils/.git/logs/refs/heads/master :: 896727e4d87c34bc5e893c2aaf76dd5b
/home/prashant/workspace/go-utils/.git/logs/refs/remotes/origin/HEAD :: 896727e4d87c34bc5e893c2aaf76dd5b
/home/prashant/workspace/go-utils/.git/logs/refs/remotes/origin/go_filehash :: 725505ae25cf563e4dbe82e4ec5e2965
/home/prashant/workspace/go-utils/.git/objects/20/3e856a9494e4da260d8ec55746817c56bdfb3f :: 099ec49f924deba8cf079f40b73ac3c2
/home/prashant/workspace/go-utils/.git/objects/25/1d18c73952886ad212156e3301b67313a5efad :: 33ffd8b72d720afe9317b013bdd59bea
/home/prashant/workspace/go-utils/.git/objects/48/08f2e7dc975bf8f0f75adbd6d1c854425a9631 :: 2c9050db064a103562551c62c445b9c9
/home/prashant/workspace/go-utils/.git/objects/4f/8199ef7361a9ccb9827906a3be1ca72cb881bd :: 33f9b251318392c56de163700aee8c67
/home/prashant/workspace/go-utils/.git/objects/66/fd13c903cac02eb9657cd53fb227823484401d :: 514fdfd11356e2e59ffb7e59e02a47d1
/home/prashant/workspace/go-utils/.git/objects/69/c894bdba209797c29d7a8991eb2db834411c4f :: 76ddb4bc138b9db30d3bfba433200b19
/home/prashant/workspace/go-utils/.git/objects/6b/4028ad7d27e0480ddb3ee77138b2b16822b967 :: df8a52bece7e01ffe8e91126c03f6396
/home/prashant/workspace/go-utils/.git/objects/be/232194908d876c206a7b90b2f955b43a665d6c :: dd5eff0ee0a21945120c0768833ccfc3
/home/prashant/workspace/go-utils/.git/objects/cd/4cbfa8960bce513255f75d6e3cbc274d73557c :: fad794b24fa83a81efaf09e1364a3469
/home/prashant/workspace/go-utils/.git/packed-refs :: ed29900ecc32b18c4e76142a96c0c304
/home/prashant/workspace/go-utils/.git/refs/heads/go_filehash :: b3a900abd651a9be893e0f70e94ea360
/home/prashant/workspace/go-utils/.git/refs/heads/master :: 2dd932ce134e10493a98cb13fed90991
/home/prashant/workspace/go-utils/.git/refs/remotes/origin/HEAD :: 73a00957034783b7b5c8294c54cd3e12
/home/prashant/workspace/go-utils/.git/refs/remotes/origin/go_filehash :: b3a900abd651a9be893e0f70e94ea360
/home/prashant/workspace/go-utils/.gitignore :: bfe95a1af03f5b904fcfd41514e12e32
/home/prashant/workspace/go-utils/README.md :: 534e546f520ae39f9f7718b1a00a2fe3
/home/prashant/workspace/go-utils/file_signatures/checksum.go :: 0c7b1f43f63ee14b42ac3f8f5a60db62
/home/prashant/workspace/go-utils/file_signatures/main.go :: 66396aa2a12b226b2abfbf7b908c407e
prashant@thinkpad:file_signatures$ 
```

### CODE REVIEW

File: file_signatures/checksum.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	"crypto/sha256"
+	"encoding/hex"
+	"hash/crc32"
+	"io"
+	"os"
+)
+
+const polynomial = 0xedb88320
+
+func FileMd5Sum(filePath string) (string, error) {
```
Comment (praju-1) [inline file_signatures/checksum.go:17]: Add header

File: file_signatures/checksum.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		return md5sum, err
+	}
+
+	hashInBytes := hash.Sum(nil)[:16]
+	md5sum = hex.EncodeToString(hashInBytes)
+
+	return md5sum, nil
+}
+
+func FileSha256(filePath string) (string, error) {
```
Comment (praju-1) [inline file_signatures/checksum.go:39]: Add header to all functions

File: file_signatures/main.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+var (
```
Comment (praju-1) [inline file_signatures/main.go:18]: What is var ? Explain

File: file_signatures/checksum.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	if _, err := io.Copy(hash, file); err != nil {
+		return shaCheckSum, err
+	}
+	hashInBytes := hash.Sum(nil)[:32]
+	shaCheckSum = hex.EncodeToString(hashInBytes)
+
+	return shaCheckSum, nil
+}
+
+func FileCrc32(filePath string) (string, error) {
```
Comment (praju-1) [inline file_signatures/checksum.go:61]: Provide docstring

**Review by praju-1 — CHANGES_REQUESTED** [review]

File: file_signatures/checksum.go (hunk 2 above)
Comment (prashant-sb) [reply file_signatures/checksum.go:39]: Done.

**Review by praju-1 — APPROVED** [review]

**Pull request merged**